src/PostgREST/Query/SqlFragment.hs

```diff
@@ -204,6 +211,17 @@ asJsonF rout
       Just r  -> (funcReturnsSingleComposite r, funcReturnsScalar r, funcReturnsSetOfScalar r)
       Nothing -> (False, False, False)
 
+asJsonArrayStripF :: Maybe Routine -> SQL.Snippet
+asJsonArrayStripF rout
+  | returnsSingleComposite    = "coalesce(json_strip_nulls(json_agg(_postgrest_t)->0), 'null')"
+  | returnsScalar             = "coalesce(json_strip_nulls(json_agg(_postgrest_t.pgrst_scalar)->0), 'null')"
+  | returnsSetOfScalar        = "coalesce(json_strip_nulls(json_agg(_postgrest_t.pgrst_scalar)), '[]')"
+  | otherwise                 = "coalesce(json_strip_nulls(json_agg(_postgrest_t)), '[]')"
+  where
+    (returnsSingleComposite, returnsScalar, returnsSetOfScalar) = case rout of
+      Just r  -> (funcReturnsSingleComposite r, funcReturnsScalar r, funcReturnsSetOfScalar r)
+      Nothing -> (False, False, False)
+
```

Comment:
Ideally we wouldn't need these 2 new functions. We could just add a Bool parameter to the existing ones to add `json_strip_nulls`.